test/plugins/features/movies/controller.test.js

```diff
@@ -4,16 +4,37 @@ const Controller = require('../../../../lib/plugins/features/movies/controller')
 
 describe('movie controller', () => {
 
-  describe('create', () => {
+  const payload = { name: 'test', release_year: 2000 };
+  let movie = {};
 
-    it('creates a movie', async () => {
-      const payload = { name: 'foo' };
+  before(async () => {
+    movie = await Controller.create(payload);
+  });
 
-      const movie = await Controller.create(payload);
+  describe('create', () => {
 
+    it('creates a movie', () => {
       expect(movie.get('name')).to.eql(payload.name);
     });
 
   });
 
+  describe('fetch', () => {
+
+    it('fetches a movie', async () => {
+      const fetchedMovie = await Controller.fetch(movie.id);
+      expect(fetchedMovie.get('name')).to.eql(payload.name);
+    });
+
+  });
+
+  describe('fetch all', () => {
+
+    it('fetches all movies', async () => {
+      const movies = await Controller.fetchAll();
+      expect(movies.length).to.eql(movie.id);
```

Comment:
I'm not understanding this assertion. Should this just be 1?

Reply:
At the start of the describe block, in the before function, a call is made to Controller.create and the response is saved, including the ID. If we assume that the Controller.create inserts the new object at the end of the table, and the ID is in increasing order, we can assume that the length of the movies table should be the same as the id of the last item in table, which in this case is the Controller.create call made by the before function.

Reply:
Change test to truncate movies db before running describe blocks.